neurodamus/node.py

```diff
@@ -940,35 +942,38 @@ def _report_build_params(self, rep_name, rep_conf, target, pop_offsets_alias_pop
         )
 
     #
-    def _report_write_coreneuron_config(self, rep_name, rep_conf, target, rep_params):
+    def _report_write_coreneuron_config(self, rep_conf, target, rep_params):
         target_spec = TargetSpec(rep_conf["Target"])
 
-        # for sonata config, compute target_type from user inputs
-        if "Sections" in rep_conf and "Compartments" in rep_conf:
-            def _compute_corenrn_target_type(section_type, compartment_type):
-                sections = ["all", "soma", "axon", "dend", "apic"]
-                compartments = ["center", "all"]
-                if section_type not in sections:
-                    raise ConfigurationError(f"Report: invalid section type '{section_type}'")
-                if compartment_type not in compartments:
-                    raise ConfigurationError(f"Report: invalid compartment type {compartment_type}")
-                if section_type == "all":  # for "all sections", support only target_type=0
-                    return 0
-                # 0=Compartment, Section { 2=Soma, 3=Axon, 4=Dendrite, 5=Apical, 6=SomaAll ... }
-                return sections.index(section_type)+1+4*compartments.index(compartment_type)
-
-            section_type = rep_conf.get("Sections")
-            compartment_type = rep_conf.get("Compartments")
-            target_type = _compute_corenrn_target_type(section_type, compartment_type)
-
-        reporton_comma_separated = ",".join(rep_params.report_on.split())
-        core_report_params = (
-            (os.path.basename(rep_conf.get("FileName", rep_name)),
-                target_spec.name, rep_params.rep_type, reporton_comma_separated)
-            + rep_params[3:5] + (target_type,) + rep_params[5:8]
-            + (target.get_gids(), SimConfig.corenrn_buff_size)
-        )
-        CoreConfig.write_report_config(*core_report_params)
+        if SimConfig.restore_coreneuron:
+            CoreConfig.update_tstop(rep_params.name, target_spec.name, rep_params.end)
+        else:
```

Comment:
Very often I deal with the "exception case" (here restore) and return. Like that the general case does not get indented with IMO makes it feel "general" and helps readability. Also the diff would be smaller. But that's just  just another way, it's perfectly fine.

Reply:
You mean to have a return after the restore? makes sense yes...
I thought to have the other else as main if and the restore in the else but i didnt like it. The return early makes more sense